extra/migrate_settings.rb

```diff
@@ -1,108 +1,200 @@
-#!/usr/bin/ruby
-#
-# Script to migrate forema-proxy settings.yml
-# to plugin form
-#
-# Greg Sutcliffe <gsutclif@redhat.com> 2014
+require 'pathname'
+require 'fileutils'
+require 'optparse'
 
-require 'yaml'
+module ::Proxy
+  class Migrations
+    attr_reader :migrations, :past_migrations, :migration_state_file_path
+    attr_accessor :executed_migrations
 
-def modules
-  [ :settings, :tftp, :dns, :dhcp, :puppet, :puppetca, :bmc, :chef, :realm ]
-end
+    def initialize(migration_state_file_path, past_migrations = [])
+      @past_migrations = past_migrations
+      @migration_state_file_path = migration_state_file_path
+      @executed_migrations = []
+    end
 
-def known_options
-  {
-    :daemon             => :settings,
-    :daemon_pid         => :settings,
-    :log_file           => :settings,
-    :log_level          => :settings,
-    :port               => :settings,
-    :ssl_ca_file        => :settings,
-    :ssl_certificate    => :settings,
-    :ssl_private_key    => :settings,
-    :trusted_hosts      => :settings,
-    :virsh_network      => :settings,
-    :foreman_url        => :settings,
-    :settings_directory => :settings,
-    :http_port          => :settings,
-    :https_port         => :settings,
-    :use_cache          => :settings,
-    :cache_location     => :settings,
-    :puppetca_use_sudo  => :puppetca,
-    :puppetdir          => :puppetca,
-    :ssldir             => :puppetca,
-    :sudo_command       => :puppetca,
-    :customrun_args     => :puppet,
-    :customrun_cmd      => :puppet,
-    :freeipa_remove_dns => :realm
-  }
-end
+    def load_past_migrations!
+      raise "Migration state file #{@migration_state_file_path} can't be found" unless File.exist?(@migration_state_file_path)
```

Comment:
I agree with @witlessbird here, since it's an input file we should raise an error when it's missing.  We can ship an empty file in the packages and mark it as a config?